File: backend/app/models/world_setting.py
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey, Enum, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base
import enum


class SettingType(str, enum.Enum):
    ERA = "era"  # 时代
    REGION = "region"  # 地域
    RULE = "rule"  # 规则（魔法/科技体系）
    CULTURE = "culture"  # 文化习俗
    POWER = "power"  # 权力结构
    LOCATION = "location"  # 地点
    FACTION = "faction"  # 势力
    ITEM = "item"  # 物品
    EVENT = "event"  # 历史事件


class WorldSetting(Base):
    __tablename__ = "world_settings"

    id = Column(Integer, primary_key=True, index=True)
    project_id = Column(Integer, ForeignKey("projects.id"), nullable=False)

    # 基础信息
    name = Column(String, nullable=False)  # 设定名称
    setting_type = Column(Enum(SettingType), nullable=False)  # 设定类型
    description = Column(Text, nullable=True)  # 详细描述

    # 扩展属性（JSON格式，灵活存储不同类型的属性）
    attributes = Column(JSON, nullable=True)  # 例如：{"level": "高级", "effect": "..."}

    # 关联关系
    related_entities = Column(JSON, nullable=True)  # 关联实体ID列表
    is_core_rule = Column(Integer, default=0)  # 是否为核心规则（不可变）

    # 图片
    image = Column(String, nullable=True)

    # 时间戳
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # 关系
    project = relationship("Project", back_populates="world_settings")

File: backend-java/src/main/java/com/novel/ai/model/entity/WorldSetting.java
package com.novel.ai.model.entity;

import java.time.LocalDateTime;

/**
 * 世界观设定实体类
 * 用于存储小说项目的世界观设定信息，包括时代、地域、规则、文化、地点、势力、物品、历史事件等
 */
public class WorldSetting {

    // 基础字段
    private Integer id;
    private Integer projectId;

    // 基础信息
    private String name;
    private String settingType;      // 设定类型：era/region/rule/culture/power/location/faction/item/event
    private String description;      // 详细描述

    // 扩展属性（JSON格式，灵活存储不同类型的属性）
    // 示例：{"level": "高级", "effect": "...", "requirements": "..."}
    private String attributes;

    // 关联关系（JSON数组格式）
    // 示例：[{"type": "character", "id": 1}, {"type": "location", "id": 2}]
    private String relatedEntities;

    // 是否为核心规则（核心规则不可变更）
    private Integer isCoreRule;

    // 图片URL
    private String image;

    // 时间戳
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // 构造函数
    public WorldSetting() {}

    public WorldSetting(Integer projectId, String name, String settingType) {
        this.projectId = projectId;
        this.name = name;
        this.settingType = settingType;
        this.isCoreRule = 0; // 默认不是核心规则
    }

    // Getter和Setter方法
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getProjectId() {
        return projectId;
    }

    public void setProjectId(Integer projectId) {
        this.projectId = projectId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSettingType() {
        return settingType;
    }

    public void setSettingType(String settingType) {
        this.settingType = settingType;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAttributes() {
        return attributes;
    }

    public void setAttributes(String attributes) {
        this.attributes = attributes;
    }

    public String getRelatedEntities() {
        return relatedEntities;
    }

    public void setRelatedEntities(String relatedEntities) {
        this.relatedEntities = relatedEntities;
    }

    public Integer getIsCoreRule() {
        return isCoreRule;
    }

    public void setIsCoreRule(Integer isCoreRule) {
        this.isCoreRule = isCoreRule;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    /**
     * 检查是否为核心规则
     */
    public boolean isCoreRule() {
        return this.isCoreRule != null && this.isCoreRule == 1;
    }
}